lib/fiber/create-instance-from-react-element.ts

```diff
@@ -69,9 +87,23 @@ const hostConfig: HostConfig<
       )
     }
 
-    const instance = prepare(new target(props) as any, {})
+    try {
+      const instance = prepare(new target(props) as any, {})
+      return instance
+    } catch (error) {
+      const rootCircuit = CircuitContext.current
 
-    return instance
+      if (rootCircuit?.db) {
+        // Add the error to the circuit-json database
+        rootCircuit.db.source_failed_to_create_component_error.insert({
+          type: "source_failed_to_create_component_error",
+          component_type: type,
+          message: error instanceof Error ? error.message : String(error),
+          error_type: "source_failed_to_create_component_error",
```

Comment:
nice, I think we could detect a zod/props error here and maybe make the error more specific e.g.

```suggestion
          error_type: "source_failed_to_create_component_error", // I can't remember why we have this, but it might be valid to use `invalid_props_for_component` as the error_type
          sub_error_type: "invalid_props_for_component",
          
          // 
          pcb_center: hasPosition ? { x: props.pcbX ?? parent.pcbX }: undefined, // Is it possible to get this? This could make it so we could display an error on the PCB /schematic so people don't wonder where their component is. It would be optional
          schematic_center: //...
```